### ISSUE TITLE
[Truffle] Correct File.read error for missing file needed for MRI Excludes.

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2836]
Expected Errno:ENOENT :
```
ruby -e 'File.read("test/ThisFileDoesNotExist.rb")'
-e:1:in `read': No such file or directory @ rb_sysopen - test/ThisFileDoesNotExist.rb (Errno::ENOENT)
	from -e:1:in `<main>'
```
Actual
```
 ./bin/jruby -X+T -e 'File.read("test/ThisFileDoesNotExist.rb")'
-e:1:in `read': internal implementation error - RuntimeException java.nio.file.NoSuchFileException: test/ThisFileDoesNotExist.rb org.jruby.truffle.runtime.util.FileUtils$1.block(FileUtils.java:39) (RubyTruffleError)
	from -e:1:in `<main>'
```

**Comment by chrisseaton** [comment]
I'm working on this but it's going to take me a while (few days) to build up the whole File stack to get to the point of implementing `File.read`.

**Comment by bjfish** [comment]
@chrisseaton Thanks for letting me know, I'm using a workaround locally in the meantime.

**Comment by chrisseaton** [comment]
Fixed in #2866
